prov/efa/src/efa_domain.c

```diff
@@ -292,6 +309,9 @@ int efa_domain_open(struct fid_fabric *fabric_fid, struct fi_info *info,
 	if (app_mr_local || !efa_mr_cache_enable)
 		return 0;
 
+	if (efa_check_fork_enabled(*domain_fid))
```

Comment:
Maybe add `efa_fork_status == EFA_FORK_SUPPORT_OFF` to the condition?

If efa_fork_status is already ON, we do not want to run the check.
If efa_fork_status is UNNEEDED, we do not want to run the check either.